### PR TITLE
Allow setting a custom placement policy per table through yb-admin_cl…

### PR DESCRIPTION
**Pull request by deeps1991**
…i that overrides the cluster placement policy.

### CODE REVIEW

**Comment by vvkgopalan** [comment]
One case to account for is when a table is colocated / part of a tablegroup, in which case setting a custom placement policy for that table should be dis-allowed. This is because colocated tables / tables within a tablegroup are placed on a singular parent tablet. For reference on implementation, see https://github.com/yugabyte/yugabyte-db/commit/6ef74d88f53dbab0af52ba793a9aaa5a7e5799a5 and https://github.com/yugabyte/yugabyte-db/commit/9fa72cd3048bacf35bc8aab7191a702ed0a6271b.

In the case of tablegroups, it may make sense to have a separate command to enable setting this placement policy per-tablegroup (so that all the tables within the tablegroup also obey it).

File: src/yb/tools/yb-admin_client.h [hunk 1]
```diff
@@ -189,6 +189,11 @@ class ClusterAdminClient {
                                 int timeout_secs,
                                 bool is_compaction);
 
+  CHECKED_STATUS ModifyTablePlacementInfo(const client::YBTableName& table_name,
+                                          const std::string& placement_info,
+                                          int replication_factor,
+                                          const std::string& optional_uuid);
```
[the remark this answers is not in the document]
Comment (deeps1991) [reply src/yb/tools/yb-admin_client.h:195]: I left it here to be future proof, but added a check to ensure that the two uuids match. Let me know if that is okay, or I can remove this in the next patch.

**Comment by deeps1991** [comment]
Thanks for the review Vivek and Rahul! I have addressed all the comments in the last commit. 
@rahuldesirazu I've implemented the change based on our offline slack discussion - and implemented checks to verify that
a) We set the custom table placement info only for a cluster with just live placement info set.
b) The placement uuid for the table matches the placement uuid of the cluster
c) We only set live placement info for a table.

Let me know if my understanding went incorrect somewhere, and I can fix those in a follow up patch.

File: src/yb/integration-tests/load_balancer_placement_policy-test.cc [hunk 2]
```diff
@@ -0,0 +1,336 @@
...
+  int rows_inserted = workload.rows_inserted();
+  LOG(INFO) << "Number of rows inserted: " << rows_inserted;
+
+  // Verify that number of rows is as expected.
+  ClusterVerifier cluster_verifier(external_mini_cluster());
+  ASSERT_NO_FATALS(cluster_verifier.CheckCluster());
+  ASSERT_NO_FATALS(cluster_verifier.CheckRowCount(
+    placement_table, ClusterVerifier::EXACTLY, rows_inserted));
+}
+
```
Comment (rahuldesirazu) [inline src/yb/integration-tests/load_balancer_placement_policy-test.cc:334]: Not sure if its already tested, but I would add/modify a test that calls ModifyPlacementInfo with the optional placement_uuid set (3rd argument) and make sure table level overrides still work.

**Review by rahuldesirazu — COMMENTED** [review]
Everything looking really solid! Just a small test case to add and a nit fix.

File: src/yb/integration-tests/load_balancer_placement_policy-test.cc (hunk 2 above)
Comment (deeps1991) [reply src/yb/integration-tests/load_balancer_placement_policy-test.cc:343]: Done

File: src/yb/client/table_alterer.h [hunk 3]
```diff
@@ -111,6 +114,10 @@ class YBTableAlterer {
 
   boost::optional<uint32_t> wal_retention_secs_;
 
+  master::ReplicationInfoPB replication_info_;
```
[the remark this answers is not in the document]
Comment (deeps1991) [reply src/yb/client/table_alterer.h]: Done

File: src/yb/integration-tests/load_balancer_placement_policy-test.cc [hunk 4]
```diff
@@ -0,0 +1,395 @@
...
+  GetLoadOnTservers(table_name().table_name(), num_tservers, &counts_per_ts);
+  ASSERT_EQ(counts_per_ts[3], 4);
+  ASSERT_EQ(counts_per_ts[4], 4);
+
+  // Modify the placement policy with different zones and replication factor but with same
+  // placement uuid.
+  ASSERT_OK(yb_admin_client_->ModifyTablePlacementInfo(
+    table_name(), "c.r.z2", 1, random_placement_uuid));
+  WaitForLoadBalancer();
+
```
[the remark this answers is not in the document]
Comment (deepthisrinivasan) [reply src/yb/integration-tests/load_balancer_placement_policy-test.cc:385]: This is actually covered in admin-cli-test

File: src/yb/integration-tests/load_balancer_placement_policy-test.cc [hunk 5]
```diff
@@ -0,0 +1,395 @@
...
+  GetLoadOnTservers(table_name().table_name(), num_tservers, &counts_per_ts);
+  // TS3 belongs to zone1 and will have 0 tablets whereas since TS4 is in zone2 it should have 4
+  // tablets allotted to it.
+  ASSERT_EQ(counts_per_ts[3], 0);
+  ASSERT_EQ(counts_per_ts[4], 4);
+
+}
+
```
[the remark this answers is not in the document]
Comment (deeps1991) [reply src/yb/integration-tests/load_balancer_placement_policy-test.cc:393]: This is covered in first few steps of PlacementPolicyTest

**Pull request merged**